server/stycoon/command_trade.go

```diff
@@ -36,8 +40,13 @@ func (game *Game) processTrade(id int64, playerOwnedShips map[int64]struct{}, pl
 		return err
 	}
 
-	if shipOwnedByPlayer || targetIsPlanet {
-		return database.ReplaceTradeCommand(game.db, id, command.Type, int64(target), int64(resource), *c.Amount)
+	err = database.CheckPlanetCanTrade(game.db, int64(target), int64(resource), int(*command.Amount))
+	if err != nil {
+		return err
+	}
+	err = database.CheckShipCanTrade(game.db, id, int64(resource), int(*command.Amount))
+	if err != nil {
+		return err
```

Comment:
Feel free to suggest a better way to structure these checks.

Reply:
Note that this mixes up `query errors` together with "logical errors".
This means that `query errors` are returned back with 400 status just like "logical errors" - it would make sense to return 500 with `query errors` but it is imho not something worth fixing.

Reply:
I agree, it would be a nice to have, let's keep it as is for now